### PR TITLE
test: add test that checks that we do not regress re/ non-standard column names

### PR DESCRIPTION
**Pull request by Taepper**
~resolves #~

### Summary

A bug was mentioned by @corneliusroemer a while back that SILO crashed on certain numeric sequence names. When I fixed this I missed to implement a test, which guarantees that we do not regress in this regard. This is that missed test

## PR Checklist

~- [ ] All necessary documentation has been adapted or there is an issue to do so.~
~- [ ] The implemented feature is covered by an appropriate test.~

### CODE REVIEW

File: testBaseData/numericNames/database_config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: version
+      type: int
+    - name: submissionId
+      type: string
+    - name: accessionVersion
+      type: string
+    - name: isRevocation
+      type: string
+    - name: versionStatus
+      type: string
```
Comment (fengelniederhammer) [inline testBaseData/numericNames/database_config.yaml:29]: Are all those fields important for the test case? If not, I would suggest to remove them (only keeping the primary key).

File: testBaseData/numericNames/reference_genomes.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      "sequence": "AAAAAAAAAAAAAAAA"
+    }
+  ],
+  "genes": [
+    {
+      "name": "someLongGene",
+      "sequence": "AAAAAAAAAAAAAAAAAAAAAAAAA"
+    },
+    {
+      "name": "someShortGene",
```
Comment (fengelniederhammer) [inline testBaseData/numericNames/reference_genomes.json:18]: Let's also have a numeric gene?

File: src/silo/preprocessing/preprocessor.test.cpp [hunk 3]
```diff
@@ -83,6 +83,29 @@ const Scenario NDJSON_WITH_SQL_KEYWORD_AS_FIELD = {
    )
 };
 
+const Scenario NDJSON_WITH_NUMERIC_NAMES = {
+   .input_directory = "testBaseData/numericNames/",
+   .expected_sequence_count = 2,
+   .query = R"(
+      {
+         "action": {
+            "type": "Aggregated",
```
Comment (fengelniederhammer) [inline src/silo/preprocessing/preprocessor.test.cpp:92]: Is this a relevant query for this test case? Let's maybe do a `Fasta` action instead?

Comment (Taepper) [reply src/silo/preprocessing/preprocessor.test.cpp:92]: The FastaAligned action cannot filter for the numeric metadata field

Comment (fengelniederhammer) [reply src/silo/preprocessing/preprocessor.test.cpp:92]: Ah, I overlooked that there is also a numeric metadata field :+1:

**Review by fengelniederhammer — APPROVED** [review]

**Pull request merged**